misc/sample_application/uservm/Makefile

```diff
@@ -1,12 +1,16 @@
-CC = gcc
-CP = g++
+CC ?= gcc
+CXX ?= g++
+T := $(CURDIR)
+OUT_DIR ?= $(shell mkdir -p $(T)i/../build;cd $(T)/../build;pwd)
 
 CFLAGS = -Wall -Wextra -Wabi=11 -pedantic
 
 LDLIBS = -lrt -pthread
 
 all: userApp.cpp
-	$(CP) $(CFLAGS) -o userapp userApp.cpp ivshmemlib.c $(LDLIBS)
+	$(CXX) $(CFLAGS) -o $(OUT_DIR)/userApp userApp.cpp ivshmemlib.c $(LDLIBS)
+	cp $(T)/histapp.py $(OUT_DIR)
```

Comment:
I don't remember you have this `cp histapp.py` command previously. Is it added intentionally? The Makefile you added under misc/sample_application does the same thing.

If it is by intention, please split the commands into two goals, one generating `$(OUT_DIR)/userApp` while the other `$(OUT_DIR)/histapp.py`. The goal `all` shall depend on those two output files rather than do everything by itself.

The idea is that if a Makefile should not rebuild something that has not changed since its last build.